platforms/nuttx/src/px4/common/print_load.cpp

```diff
@@ -77,10 +84,7 @@ void init_print_load(struct print_load_s *s)
 	s->new_time = system_load.start_time;
 	s->interval_start_time = system_load.start_time;
 
-	sched_lock();
-	// special case for IDLE thread
-	s->last_times[0] = system_load.tasks[0].total_runtime;
```

Comment:
I think a better fix would be to just set `s->last_times[i] = system_load.tasks[i].total_runtime` for all tasks, and also set `new_time` and `interval_start_time` correctly. Then it should work independently from whether measurement just started or is already running.